homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,120 @@
+"""Support for Broadlink climate devices."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Broadlink climate entities."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type in {"Hysen heating controller"}:
+        climate_entities = [BroadlinkHysen(device)]
+    else:
+        _LOGGER.warning("Unrecognised broadlink climate API.")
+    async_add_entities(climate_entities)
+
+
+class BroadlinkHysen(ClimateEntity):
```

Comment:
We need to subclass RestoreEntity to make the state persistent. [Example](https://github.com/home-assistant/core/blob/f312b87a3f90681d5f1a7fda99939449c2fa0766/homeassistant/components/broadlink/remote.py#L111-L112).